src/core/thread.d

```diff
@@ -2293,7 +2294,7 @@ extern (C) void thread_joinAll()
             Thread.remove(t);
             t = tn;
         }
-        else if (t.isDaemon)
```

Comment:
Perhaps the synchronization can be done with something else that doesn't rely on the GC to clean up the Mutex objects.